hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -533,6 +540,18 @@ public static Stream<HoodieLogFile> getAllLogFiles(FileSystem fs, Path partition
     }
   }
 
+  public static Stream<HoodieLogFile> getAllLogFiles(FileSystem fs, String partitionPath, final String fileId,
+      final String logFileExtension, final String baseCommitTime, HoodieStorageStrategy storageStrategy) throws IOException {
```

Comment:
high-level design suggestion: there are quite some API calls that pass strategy objects around. i think we need a centralized place providing APIs to retrieve files so that callers do not need to know about the storage strategy. maybe make use of FSView.

Reply:
This makes sense to me. I'll investigate to see how we can avoid passing objects around